tests/integration/test_random_bitgenerator.py

```diff
@@ -192,6 +237,95 @@ def test_poisson(t):
+def test_random_size_none(func):
+    seed = 42
+    gen_np = np.random.Generator(np.random.PCG64(seed=seed))
+    gen_num = num.random.Generator(num.random.XORWOW(seed=seed))
+    a_np = getattr(gen_np, func)(size=None)
+    a_num = getattr(gen_num, func)(size=None)
+    # cuNumeric returns singleton array
+    # NumPy returns scalar
+    assert np.ndim(a_np) == np.ndim(a_num)
+
+
+class TestBitGeneratorErrors:
+    def test_init_bitgenerator(self):
+        expected_exc = NotImplementedError
+        with pytest.raises(expected_exc):
+            np.random.BitGenerator()
+        with pytest.raises(expected_exc):
+            num.random.BitGenerator()
+
+    @pytest.mark.xfail
+    @pytest.mark.parametrize("dtype", (np.int32, np.float128, str))
+    def test_random_invalid_dtype(self, dtype):
+        expected_exc = TypeError
+        seed = 42
+        gen_np = np.random.Generator(np.random.PCG64(seed=seed))
+        gen_num = num.random.Generator(num.random.XORWOW(seed=seed))
+        with pytest.raises(expected_exc):
+            gen_np.random(size=(1024 * 1024,), dtype=dtype)
+            # TypeError: Unsupported dtype dtype('int32') for random
+        with pytest.raises(expected_exc):
+            gen_num.random(size=(1024 * 1024,), dtype=dtype)
+            # NotImplementedError: type for random.uniform has to be float64
+            # or float32
+
+    @pytest.mark.xfail
+    def test_random_out_dtype_mismatch(self):
+        expected_exc = TypeError
+        seed = 42
+        dtype = np.float32
+        out_shape = (3, 2, 3)
+        out_dtype = np.float64
+        gen_np = np.random.Generator(np.random.PCG64(seed=seed))
+        gen_num = num.random.Generator(num.random.XORWOW(seed=seed))
+        out_np = np.empty(out_shape, dtype=out_dtype)
+        out_num = num.empty(out_shape, dtype=out_dtype)
+        with pytest.raises(expected_exc):
+            gen_np.random(out=out_np, dtype=dtype)
+            # TypeError: Supplied output array has the wrong type.
+            # Expected float32, got float64
+        with pytest.raises(expected_exc):
+            gen_num.random(out=out_num, dtype=dtype)
+            # assert out.dtype == dtype
+            # AssertionError
+
+    @pytest.mark.xfail
+    def test_random_out_shape_mismatch(self):
+        expected_exc = ValueError
+        seed = 42
+        size = (1024 * 1024,)
+        out_shape = (3, 2, 3)
+        gen_np = np.random.Generator(np.random.PCG64(seed=seed))
+        gen_num = num.random.Generator(num.random.XORWOW(seed=seed))
+        out_np = np.empty(out_shape)
+        out_num = num.empty(out_shape)
+        with pytest.raises(expected_exc):
+            gen_np.random(size=size, out=out_np)
+            # ValueError: size must match out.shape when used together
+        with pytest.raises(expected_exc):
+            gen_num.random(size=size, out=out_num)
+            # assert out.shape == size
+            # AssertionError
```

Comment:
@manopapad same question: is this cunumeric being more general, or something we should add a check for to match the np exception?

Reply:
Same here, mis-using assertions to check user input.

Reply:
@yimoj can you replace the assertion with a `raise TypeError` so that this test (and the one below) can pass? That's a small enough change that it would be less overhead to just fix it here in this PR, I think.